### PR TITLE
Add notifications when changing privacy preferences

### LINKED ISSUE #3253: no notifications when changing privacy settings
I thought we had these.

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/9635333-no-notifications-when-changing-privacy-settings?utm_campaign=plugin&utm_content=tracker%2F85909&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F85909&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by rohitpaulk**
fixes #3253

### CODE REVIEW

**Comment by rohitpaulk** [comment]
I'm thinking of combining the `anonymous.json` and `privacy.json` endpoints into one. They aren't mentioned anywhere in our API documentation, so I guess this shouldn't be a problem?

**Comment by Changaco** [comment]
> I'm thinking of combining the `anonymous.json` and `privacy.json` endpoints into one.

Seems like a good refactoring to me, go for it.

File: www/%username/settings/index.html.spt [hunk 1]
```diff
@@ -171,25 +171,25 @@ emails = participant.get_emails()
...
-            {% endif %}
             <br />
+            {% endif %}
             <label class="is-searchable">
-                <input type="checkbox"
-                    {% if not participant.is_searchable %}checked="true"{% endif %} />
-                    {{ _("Hide myself from search results.") }}
+                <input type="checkbox" data-field="is_searchable"
+                    {% if participant.is_searchable %}checked="true"{% endif %} />
+                    {{ _("Show myself in search results.") }}
```
Comment (Changaco) [inline www/%username/settings/index.html.spt]: Why did you change that?

Comment (rohitpaulk) [inline www/%username/settings/index.html.spt]: I didn't want to complicate the JS with a specific check for `is_searchable`. With this change, it just uses the value of the field to update the `checked` attribute of the input.

`$(e.target).attr('checked', data[field]);`

Comment (Changaco) [inline www/%username/settings/index.html.spt]: Added a commit to restore this.
